public/js/app.js

```diff
@@ -0,0 +1,110 @@
+// Call function getNextMove(player, coord)
+// player = 'computer' or 'human'
+// coord is an element in [0,1,2,3,4,5,6,7,8]
+// The function returns an element in [0,1,2,3,4,5,6,7,8]
+// as the computer's next move, 'computer', 'human', or 'tie'
+
+computerMoves = []
+humanMoves = []
+board = []
+wins = [[0,1,2],[3,4,5],[6,7,8],[0,3,6],[1,4,7],[2,5,8],[0,4,8],[2,4,6]]
+
+function getNextMove(player, coord){
+    _updateBoard(player, coord);
+    var nextMove = _nextMove(_otherPlayer(player))[0];
+    _updateBoard(_otherPlayer(player), nextMove);
+    console.log(nextMove);
+}
+
+function _updateBoard(player, coord) {
+    board.push(coord);
+    if (player == 'computer')
+        computerMoves.push(coord);
+    else
+        humanMoves.push(coord);
+}
+
+function _undoBoard(player, coord) {
+    board.pop();
+    if (player == 'computer')
+        computerMoves.pop();
+    else
+        humanMoves.pop();
+}
+
+function _otherPlayer(player) {
+    if (player == 'computer')
+        return 'human';
+    else
+        return 'computer';
+}
+
+function _isSubArray (subArray, Array) {
+    for(var i = 0; i < subArray.length; i++) {
+        if($.inArray(subArray[i], Array) == -1)
+            return false;
+    }
+    return true;
+}
+
+function _boardStatus() {
+    for(var i = 0; i < wins.length; i++) {
+        if (_isSubArray(wins[i], computerMoves))
+            return 'computer';
+        else if (_isSubArray(wins[i], humanMoves))
+            return 'human';
+    }
+    if (board.length == 9)
+        return 'tie';
+    else
+        return 'game is still on'
+}
+
+function _getAvailableMoves() {
+    return $.grep([0,1,2,3,4,5,6,7,8], function(n,i) {
+        return $.inArray(n, board) == -1});;
+}
+
```

Comment:
This is the meat of your app logic, in which you basically play the keep playing the game until you find a winner.  Just keep in mind the advantages and disadvantages of this approach.  One key disadvantage is that you potentially have to play out all possibilities to get your next move, on another game with a board of over 100 positions, this will be too costly.

Lower in priority:
Also, you only have an _unBoard method because you use the actual board to run your possible outcomes on.  Its best to encapsulate your data structures in addition to your methods, so that its less bug prone and easier to debug.  It the game doesn't work as intended, you couldn't safely rule out that the actual board isn't permanently being altered when you play out the possible moves.  I suggest making a copy of the board and playing out the possibilities on the copy.  You might have to change your data structure logic.  Would it be possible to base your game logic on a board like this ['P', null, 'P', 'C', null, null, null, null] without the need to have 2 additional arrays for comp and player moves?